### ISSUE TITLE
[QUESTION] How to enable auto-completion in shell by default

### ISSUE THREAD

**Issue by WhatDothLife** [hlissner/doom-emacs#1926]
Everytime I launch the shell interface via `M-: shell`, auto-completion isn't enabled immediately. Strangely, company-mode is still set to t, but I have to first run `doom//refresh` to make it work. What could be the reason for this behavior?

My `M-: doom/info`:

<details>

```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD LCMS2
        build      Sep 16, 2019
        buildopts  (--build=x86_64-linux-gnu --prefix=/usr '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' --sysconfdir=/etc --localstatedir=/var --disable-silent-rules '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' --disable-maintainer-mode --disable-dependency-tracking --prefix=/usr --sharedstatedir=/var/lib --program-suffix=26 --with-modules --with-file-notification=inotify --with-mailutils --with-x=yes --with-x-toolkit=gtk3 --with-xwidgets --with-lcms2 'CFLAGS=-g -O2 -fdebug-prefix-map=/build/emacs26-TP6iDo/emacs26-26.3~1.git96dd019=. -fstack-protector-strong -Wformat -Werror=format-security -no-pie' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -no-pie')
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop, origin/develop, origin/HEAD 4998b5684 2019-10-17 20:08:47 -0400
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /bin/bash
        uname      Linux 5.0.0-31-generic #33~18.04.1-Ubuntu SMP Tue Oct 1 10:20:39 UTC 2019 x86_64
        path       (/usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games /snap/bin /usr/lib/x86_64-linux-gnu/emacs/26.3/x86_64-linux-gnu)
config  envfile    nil
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash ophints (popup +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces :editor (evil +everywhere) file-templates fold format multiple-cursors rotate-text :emacs dired electric :term term :emacs vc :tools eval flycheck flyspell (lookup +docsets) magit make pass pdf :lang cc data emacs-lisp ess lua markdown (org +attach +babel +capture +export +habit +present +protocol) sh :email mu4e :config (default +bindings +smartparens))
        packages   ((csv-mode :disable t) (auctex :disable t) (xclip :disable t) (gnu-elpa-keyring-update :disable t) (rustic :disable t) (adaptive-wrap :disable t) (company-auctex :disable t) exwm xelb academic-phrases ess buffer-move symon smtpmail)
        elpa       (n/a)
```
</details>

**Comment by WhatDothLife** [comment]
I'm using the bash.

**Comment by hlissner** [comment]
Please include your `M-x doom/info`, as well as which terminal emulator you're using (Doom has four, for term, vterm, shell and eshell), all of which rely on different mechanics for completion. If you don't know, run `M-: major-mode` in the terminal buffer and report to me its return value.

Also run `M-: company-backends` and let me know what it returns.

> I have to first run doom//refresh to make it work

That's no surprise. `doom//refresh` reloads the entire config. I don't recommend you use it, as it's the nuclear option.

**Comment by WhatDothLife** [comment]
For me `M-: company-backends` isn't available for interactive use.

**Comment by hlissner** [comment]
Then use `M-x eval-expression RET company-backends` instead (same for the other commands).

**Comment by WhatDothLife** [comment]
Thanks for the advice. It returns `(company-clang)`

**Comment by hlissner** [comment]
And the rest?

**Comment by WhatDothLife** [comment]
I use the emacs shell interface `M-: shell`. I launch it with the latter command or `SPC-a-s`.

**Comment by WhatDothLife** [comment]
[Here](https://github.com/WhatDothLife/doom-config) is the private config.

**Comment by WhatDothLife** [comment]
The issue was solved by removing all the Emacs-generated code from my init.el. This also solved #1995
